db/columns.py

```diff
@@ -0,0 +1,54 @@
+from sqlalchemy import Column, Integer
+from db import constants
+
+
+PRIMARY_KEY = "primary_key"
+TYPE = "type"
+
+DEFAULT_COLUMNS = {
+    constants.ID: {TYPE: Integer, PRIMARY_KEY: True}
+}
+
+
+# TODO replicate ForeignKey wrangling from prototype
+class MathesarColumn(Column):
+    """
+    This class constrains the possible arguments, enabling us to include a copy
```

Comment:
@mathemancer what arguments do we not accept any more and why was the copy method deprecated upstream?

Reply:
The idea is to show what arguments _are_ accepted to the caller via the arguments to the `__init__` function.  As we expand the need for functionality, and implement the needed copying logic, we can add to those args.  For context, the SA `Column` object accepts 15+ kwargs.  We don't need most (almost any) of them.

The upstream `copy` method was deprecated because it can't faithfully copy all the details of the SA `Column` object, leading to unexpected behavior.  This is the motivation for restricting our definition of a `MathesarColumn` to only define the parts of an SA `Column` that we'll actually need (and extending if needed later); it'll make `copy` easier to implement.

Reply:
I think this would be good context to have in the code comments.

Reply:
I added some docstring notes to (hopefully) make this clearer.